### PR TITLE
Show AddUserSecrets without CreateDefaultBuilder

### LINKED ISSUE #17892: 3.1 update of Safe storage of app secrets i
When CreateDefaultBuilder isn't called, add the user secrets configuration source explicitly by calling AddUserSecrets in the Startup constructor.

The constructor is:
public Startup(IConfiguration configuration)
    {
        Configuration = configuration;
    }

where this is from:
public Startup(IWebHostEnvironment env){...}

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9c99993c-c983-6fda-6c3a-ce4ec8c1d5db
* Version Independent ID: ba688ac2-e90e-fce5-f60d-40c2553c3efd
* Content: [Safe storage of app secrets in development in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/security/app-secrets?view=aspnetcore-3.1&tabs=windows)
* Content Source: [aspnetcore/security/app-secrets.md](https://github.com/dotnet/AspNetCore.Docs/blob/master/aspnetcore/security/app-secrets.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-security**
* GitHub Login: @Rick-Anderson
* Microsoft Alias: **scaddie**

### PR DESCRIPTION
**Pull request by serpent5**
Fixes #17892.

### CODE REVIEW

File: aspnetcore/security/app-secrets.md [hunk 1]
```diff
@@ -69,7 +69,7 @@ dotnet user-secrets init
 
 The preceding command adds a `UserSecretsId` element within a `PropertyGroup` of the *.csproj* file. By default, the inner text of `UserSecretsId` is a GUID. The inner text is arbitrary, but is unique to the project.
 
-[!code-xml[](app-secrets/samples/2.x/UserSecrets/UserSecrets.csproj?name=snippet_PropertyGroup&highlight=3)]
+[!code-xml[](app-secrets/samples/3.x/UserSecrets/UserSecrets.csproj?name=snippet_PropertyGroup&highlight=3)]
```
Comment (serpent5) [inline aspnetcore/security/app-secrets.md:72]: A few of the code references in this 3.x section were referencing the 2.x samples, so I updated the links and added the relevant code.

File: aspnetcore/security/app-secrets.md [hunk 2]
```diff
@@ -136,18 +136,17 @@ Open a command shell, and execute the following command:
 
 The [ASP.NET Core Configuration API](xref:fundamentals/configuration/index) provides access to Secret Manager secrets.
 
-In ASP.NET Core 2.0 or later, the user secrets configuration source is automatically added in development mode when the project calls <xref:Microsoft.AspNetCore.WebHost.CreateDefaultBuilder%2A> to initialize a new instance of the host with preconfigured defaults. `CreateDefaultBuilder` calls <xref:Microsoft.Extensions.Configuration.UserSecretsConfigurationExtensions.AddUserSecrets%2A> when the <xref:Microsoft.AspNetCore.Hosting.IHostingEnvironment.EnvironmentName> is <xref:Microsoft.AspNetCore.Hosting.EnvironmentName.Development>:
+The user secrets configuration source is automatically added in development mode when the project calls <xref:Microsoft.Extensions.Hosting.Host.CreateDefaultBuilder%2A> to initialize a new instance of the host with preconfigured defaults. `CreateDefaultBuilder` calls <xref:Microsoft.Extensions.Configuration.UserSecretsConfigurationExtensions.AddUserSecrets%2A> when the <xref:Microsoft.Extensions.Hosting.IHostEnvironment.EnvironmentName> is <xref:Microsoft.Extensions.Hosting.EnvironmentName.Development>:
```
Comment (serpent5) [inline aspnetcore/security/app-secrets.md:139]: This was referencing `WebHost.CreateDefaultBuilder`, even though it shows `Host.CreateDefaultBuilder`. I fixed that and the other references for the 3.x changes.

I couldn't review this, as the xrefs aren't processed, but I hope I got things right.

File: aspnetcore/security/app-secrets.md [hunk 3]
```diff
@@ -382,15 +381,13 @@ The [ASP.NET Core Configuration API](xref:fundamentals/configuration/index) prov
...
 
 [!code-csharp[](app-secrets/samples/2.x/UserSecrets/Program.cs?name=snippet_CreateWebHostBuilder&highlight=2)]
 
-
 When `CreateDefaultBuilder` isn't called, add the user secrets configuration source explicitly by calling <xref:Microsoft.Extensions.Configuration.UserSecretsConfigurationExtensions.AddUserSecrets%2A> in the `Startup` constructor. Call `AddUserSecrets` only when the app runs in the Development environment, as shown in the following example:
 
-[!code-csharp[](app-secrets/samples/1.x/UserSecrets/Startup.cs?name=snippet_StartupConstructor&highlight=12)]
+[!code-csharp[](app-secrets/samples/2.x/UserSecrets/Startup3.cs?name=snippet_StartupConstructor&highlight=12)]
```
Comment (serpent5) [inline aspnetcore/security/app-secrets.md:390]: This was the only referenced 1.x file, so I fixed it up to reference 2.x and killed off all the 1.x files we no longer need.

File: aspnetcore/security/app-secrets/samples/3.x/UserSecrets/Program2.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+                {
+                    // Add other providers for JSON, etc.
+
+                    if (hostContext.HostingEnvironment.IsDevelopment())
+                    {
+                        builder.AddUserSecrets<Program>();
+                    }
+                })
+                .Build();
+                #endregion
```
Comment (serpent5) [inline aspnetcore/security/app-secrets/samples/3.x/UserSecrets/Program2.cs:23]: This is the new stuff. I simplified from the snippet I pasted into the linked issue, to make it more consistent with [this approach](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/generic-host?view=aspnetcore-2.1#shutdown-timeout).

**Comment by Rick-Anderson** [comment]
Nice work.

**Pull request merged**